Cargo.toml

```diff
@@ -113,6 +114,13 @@ noirc_abi = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-
 noirc_artifacts = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
 noirc_printable_type = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
 bn254_blackbox_solver = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+nargo = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+nargo_toml = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+noirc_driver = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+noirc_errors = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+fm = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
+nargo_cli = { git = "https://github.com/noir-lang/noir", rev = "nightly-2025-05-28" }
```

Comment:
I wasn't expecting the command line specific stuff to be linked in. Fine for now, but we probably want to trim this before audit.

Reply:
This is mostly for the `compile_workspace` funciton so we can avoid calling the external `nargo` binary in tests & benchmarks